nestedtensor/csrc/nested_tensor_impl.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <nestedtensor/csrc/nested_tensor.h>
+
+namespace at {
+
+constexpr auto NestedTensorKey = DispatchKey::PrivateUse1_PreAutograd;
+
+struct NestedTensorImpl : public c10::TensorImpl {
+  explicit NestedTensorImpl(torch::nested_tensor::NestedTensor&& data)
+      : TensorImpl(
+            c10::DispatchKeySet(NestedTensorKey),
+            data.dtype(),
+            data.device()),
+        _data(std::move(data)) {}
+
+  int64_t dim() const override {
+    return _data.dim();
+  }
+  int64_t numel() const override {
+    return _data.numel();
+  }
+  bool is_contiguous(
+      at::MemoryFormat memory_format) const override {
+    return _data.is_contiguous();
+  }
+
+  IntArrayRef sizes() const override;
+  int64_t size(int64_t dim) const override;
+  IntArrayRef strides() const override;
+
+  torch::nested_tensor::NestedTensor _data;
+
+};
+
+inline torch::nested_tensor::NestedTensor get_nested_tensor(
```

Comment:
If we are constructing a new Nested Tensor, i would rename this method to "construct_nested_tensor" or something similar

Reply:
Also, should we be returning a pointer?

Reply:
Sure we can pick a different name. Why would we return a pointer?